public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -110,26 +112,35 @@ export class PanelQueryRunner {
       return of(snapshotPanelData);
     }
 
-    return this.subject.pipe(
-      mergeMap((data: PanelData) => {
+    const panel = this.dataConfigSource as unknown as PanelModel;
+    let dashData: Observable<DashboardQueryRunnerResult>;
+
+    try {
+      dashData = getDashboardQueryRunner().getResult(panel.id);
+    } catch (err) {
+      dashData = emptyResult();
```

Comment:
Technically we wouldn't ever get in to this catch handler unless we forgot to init the dashboard query runner in a test or something

Reply:
why are we joining in the dashboard query result here again? it is already joined with the query result here:
https://github.com/grafana/grafana/blob/bf197d2f0435602a8d70ba2c9b57207c6ffe2313/public/app/features/query/state/PanelQueryRunner.ts#L372